integration-tests/src/lib.rs

```diff
@@ -313,7 +316,19 @@ fn reserve_transfer_native_asset_from_para_to_system_para() {
 	let destination = PopNetworkPara::sibling_location_of(AssetHubPara::para_id());
 	let beneficiary_id = AssetHubParaReceiver::get(); // bob on asset hub
 	let amount_to_send = PopNetworkPara::account_data_of(PopNetworkParaReceiver::get()).free; // bob on pop balance
-	let assets = (Parent, amount_to_send).into();
+	let assets: Assets = (Parent, amount_to_send).into();
+
+	let delivery_fees = PopNetworkPara::execute_with(|| {
+		xcm_helpers::teleport_assets_delivery_fees::<
+			<PopNetworkXcmConfig as xcm_executor::Config>::XcmSender,
+		>(
+			assets.clone(), 0, Unlimited, Location::from(beneficiary_id.clone()), destination.clone()
+		)
+	});
+	#[cfg(feature = "mainnet")]
+	let amount_to_send = amount_to_send - (ASSET_HUB_ED + delivery_fees);
+	#[cfg(feature = "mainnet")]
+	let assets: Assets = (Parent, amount_to_send).into();
```

Comment:
Wouldn't it work if we don't check those variables before you calculate the delivery_fees?
```suggestion
	let amount_to_send = amount_to_send - (ASSET_HUB_ED + delivery_fees);
	let assets: Assets = (Parent, amount_to_send).into();
```